### PR TITLE
dotCMS/core#19353 Create new logout page for SAML

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/api/services/dot-router/dot-router.service.ts [hunk 1]
```diff
@@ -86,7 +87,12 @@ export class DotRouterService {
             : this.redirectMain();
     }
 
-    goToLogin(parameters?: any): void {
+    goToLogin(parameters?: NavigationExtras): void {
+        if (parameters) {
```
Comment (hmoreras) [inline src/app/api/services/dot-router/dot-router.service.ts:91]: Still need to double check if this is needed.

File: src/app/api/services/dot-router/dot-router.service.spec.ts [hunk 2]
```diff
@@ -178,12 +179,37 @@ describe('DotRouterService', () => {
...
+        expect(router.navigate).toHaveBeenCalledWith([], {
+            queryParams: params,
+            queryParamsHandling: 'merge'
+        });
+    });
+
+    describe('go to login', () => {
+        beforeEach(() => {
+            const mockDate = new Date(1466424490000);
+            jasmine.clock().install();
```
Comment (fmontes) [inline src/app/api/services/dot-router/dot-router.service.spec.ts:193]: 😯 didn't know this.

File: src/app/view/components/_common/iframe/iframe-component/iframe.component.spec.ts [hunk 3]
```diff
@@ -1,13 +1,12 @@
 import { RouterTestingModule } from '@angular/router/testing';
```
Comment (fmontes) [inline src/app/view/components/_common/iframe/iframe-component/iframe.component.spec.ts:1]: What happen here?

File: src/app/view/components/_common/iframe/iframe-component/iframe.component.ts [hunk 4]
```diff
@@ -210,7 +209,7 @@ export class IframeComponent implements OnInit, OnDestroy {
     private handleErrors(error: number): void {
         const errorMapHandler = {
             401: () => {
-                this.loginService.logOutUser().subscribe(_data => {});
+               window.location.href = '/dotAdmin/logout';
```
Comment (fmontes) [inline src/app/view/components/_common/iframe/iframe-component/iframe.component.ts:212]: can we test this?

File: src/app/view/components/dot-toolbar/components/dot-my-account/dot-my-account.component.ts [hunk 5]
```diff
@@ -127,7 +127,7 @@ export class DotMyAccountComponent implements OnInit, OnDestroy {
                 this.close.emit();
 
                 if (response.entity.reauthenticate) {
-                    this.loginService.logOutUser().subscribe(() => {});
+                    window.location.href = '/dotAdmin/logout';
```
Comment (fmontes) [inline src/app/view/components/dot-toolbar/components/dot-my-account/dot-my-account.component.ts:130]: Should we do this in the router service?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: src/app/api/services/dot-router/dot-router.service.spec.ts [hunk 6]
```diff
@@ -38,29 +38,31 @@ describe('DotRouterService', () => {
...
-                },
-                {
-                    provide: ActivatedRoute,
-                    useClass: ActivatedRouteMock
-                }
-            ],
-            imports: [RouterTestingModule]
-        });
+    beforeEach(
+        async(() => {
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-router/dot-router.service.spec.ts:42]: remove

File: src/app/api/services/dot-router/dot-router.service.spec.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply src/app/api/services/dot-router/dot-router.service.spec.ts:193]: what's that for?

File: src/app/view/components/dot-toolbar/components/dot-my-account/dot-my-account.component.spec.ts [hunk 7]
```diff
@@ -235,15 +235,13 @@ describe('DotMyAccountComponent', () => {
         const changePassword = de.query(By.css('#dot-my-account-change-password-option'));
```
Comment (alfredo-dotcms) [inline src/app/view/components/dot-toolbar/components/dot-my-account/dot-my-account.component.spec.ts:235]: migrate this to `TestBed`

File: src/app/view/components/login/forgot-password-component/forgot-password.component.ts [hunk 8]
```diff
@@ -72,10 +78,7 @@ export class ForgotPasswordComponent implements OnInit {
      *
      * @memberof ForgotPasswordComponent
```
Comment (alfredo-dotcms) [inline src/app/view/components/login/forgot-password-component/forgot-password.component.ts:79]: update doc

File: src/app/api/services/dot-router/dot-router.service.ts [hunk 9]
```diff
@@ -86,8 +87,8 @@ export class DotRouterService {
             : this.redirectMain();
     }
 
-    goToLogin(parameters?: any): void {
-        this.router.navigate(['/public/login'], parameters);
+    goToLogin(navExtras?: NavigationExtras): void {
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-router/dot-router.service.ts:90]: doc?

File: src/app/api/services/dot-router/dot-router.service.spec.ts (hunk 2 above)
Comment (hmoreras) [reply src/app/api/services/dot-router/dot-router.service.spec.ts:193]: to mock the value of `new Date()`

**Pull request merged**